pydy/viz/server.py

```diff
@@ -2,36 +2,130 @@
 
 import os
 import sys
+import signal
+import socket
 import webbrowser
+
+# For python 2 and python 3 compatibility
 if sys.version_info < (3, 0):
     from SimpleHTTPServer import SimpleHTTPRequestHandler
     from BaseHTTPServer import HTTPServer
 else:
     from http.server import SimpleHTTPRequestHandler
     from http.server import HTTPServer
+    raw_input = input
+
+__all__ = ['Server']
+
+
+class StoppableHTTPServer(HTTPServer):
+    """
+    Overrides BaseHTTPServer.HTTPServer to include a stop
+    function.
+    """
+
+    def server_bind(self):
+        HTTPServer.server_bind(self)
+        self.socket.settimeout(1)
+        self.run = True
+
+    def get_request(self):
+        while self.run:
+            try:
+                sock, addr = self.socket.accept()
+                sock.settimeout(None)
+                return (sock, addr)
+            except socket.timeout:
+                pass
+
+    def stop(self):
+        self.run = False
+
+    def serve(self):
+        while self.run:
+            try:
+                self.handle_request()
+            except TypeError:
+                # When server is being closed, while loop can run once
+                # after setting self.run = False depending on how it
+                # is scheduled.
+                pass
+
+
+class Server(object):
+    """
+    Parameters
+    ----------
+    port : integer
+        Defines the port on which the server will run. If this port is
+        already bind, then it increment 1 until it finds a free port.
+    scene_file : name of the scene_file generated for visualization
+        A Valid PyDy generated scene file in 'directory' parameter.
+    directory : absolute path of a directory
+        Absolute path to the directory which contains scene_file with
+        all other static files.
+
+    Example
+    -------
+    >>> server = Server(scene_file=_scene_json_file)
+    >>> server.run_server()
+
+    """
+    def __init__(self, scene_file, directory=None, port=None):
+        self.scene_file = scene_file
 
+        if directory is None:
+            self.directory = directory
+        else:
+            self.directory = "static/"
 
-__all__ = ['run_server']
+        if port is None:
```

Comment:
But, I don't see how using `directory='static'` is a good idea. That's why I suggested using `directory=None` and then if that was true setting `directory = os.getcwd()`.